sormas-ui/src/main/java/de/symeda/sormas/ui/samples/PathogenTestController.java

```diff
@@ -161,120 +162,238 @@ public static void showCaseUpdateWithNewDiseaseVariantDialog(CaseDataDto existin
 			onSavedPathogenTest.accept(dto, () -> {
-				confirmCaseCallback.run();
-				caseCloningCallback.run();
-				caseDiseaseVariantCallback.run();
+				positiveTestCallback.run();
+				negativeTestCallback.run();
 			});
 		} else {
-			confirmCaseCallback.run();
-			caseCloningCallback.run();
-			caseDiseaseVariantCallback.run();
+			positiveTestCallback.run();
+			negativeTestCallback.run();
 		}
 	}
 
 	private void handleAssociatedContact(
 		PathogenTestDto dto,
 		BiConsumer<PathogenTestDto, Runnable> onSavedPathogenTest,
 		ContactReferenceDto associatedContact) {
+
+		// Negative test result AND test result verified
+		// a) Tested disease == contact disease AND test result != sample pathogen test result: Ask user whether to update the sample pathogen test result
+		// b) Tested disease != contact disease: Do nothing
+
+		// Positive test result AND test result verified
+		// a) Tested disease == contact disease: Ask user to convert the contact to a case
+		// a.1) If contact is converted, update the sample pathogen test result
+		// a.2) If contact is not converted (or there already is a resulting case), ask user whether to update the sample pathogen test result
+		// b) Tested disease != contact disease: Ask user to create a new case for the tested disease
+
 		final ContactDto contact = FacadeProvider.getContactFacade().getContactByUuid(associatedContact.getUuid());
-		Runnable contactConvertToCaseCallback = () -> {
-			if (PathogenTestResultType.POSITIVE.equals(dto.getTestResult())
-				&& dto.getTestResultVerified()
-				&& ContactClassification.UNCONFIRMED.equals(contact.getContactClassification())
-				&& !ContactStatus.CONVERTED.equals(contact.getContactStatus())) {
-				if (contact.getDisease() != null && contact.getDisease().equals(dto.getTestedDisease())) {
-					showConvertContactToCaseDialog(contact);
-				} else if (contact.getDisease() != null) {
+		final boolean equalDisease = dto.getTestedDisease() == contact.getDisease();
+
+		Runnable negativeTestCallback = () -> {
+			if (equalDisease && PathogenTestResultType.NEGATIVE.equals(dto.getTestResult()) && dto.getTestResultVerified()) {
+				changeAssociatedSampleResult(dto, handleChanges -> {
+				});
+			}
+		};
+
+		Runnable positiveTestCallback = () -> {
+			if (PathogenTestResultType.POSITIVE.equals(dto.getTestResult()) && dto.getTestResultVerified()) {
+				if (equalDisease) {
+					if (!ContactStatus.CONVERTED.equals(contact.getContactStatus())) {
+						showConvertContactToCaseDialog(contact, converted -> {
+							handleCaseCreationFromContactOrEventParticipant(converted, dto);
+						});
+					} else {
+						changeAssociatedSampleResult(dto, handleChanges -> {
+						});
+					}
+				} else {
 					showCreateContactCaseDialog(contact, dto.getTestedDisease());
 				}
 			}
 		};
+
 		if (onSavedPathogenTest != null) {
-			onSavedPathogenTest.accept(dto, contactConvertToCaseCallback);
+			onSavedPathogenTest.accept(dto, () -> {
+				positiveTestCallback.run();
```

Comment:
Technically, there is no need to provide two separate callbacks here for positive and negative test results.
However, it is okay if it is supposed to be more readable like this.